core-rust/state-manager/src/state_manager.rs

```diff
@@ -837,14 +790,93 @@ where
         final_ledger_proof
     }
 
+    fn execute_genesis_scenario(
+        &self,
+        genesis_commit_request_factory: &mut GenesisCommitRequestFactory,
+        scenario_name: &str,
+        epoch: Epoch,
+        nonce: u32,
+    ) -> u32 {
+        let mut scenario = self
+            .find_scenario(epoch, nonce, scenario_name)
+            .unwrap_or_else(|| {
+                panic!(
+                    "Could not find scenario with logical name: {}",
+                    scenario_name
+                )
+            });
+        let scenario_base_version = genesis_commit_request_factory.state_version;
```

Comment:
I think this is a bit confusing that it is the state version _before_ the scenario executed.

I'd prefer to just set this to be:
```rust
    if let Some(scenario_start_state_version) = committed_transactions.first() {
        // We only commit a record of the scenario if it commits at least one transaction
        self.store.write().put(scenario_start_state_version, executed_scenario);
    }
```

Reply:
TBH a "state version after the first transaction that got committed as part of the scenario" is also a bit arbitrary to me :(
And in both cases, "not storing information on attempted scenarios which didn't successfully commit transactions" is also a bit imperfect.

Sooo... why do we want to identify these scenarios by a _transaction's_ ID anyway? Can we do "scenario order number" instead? Like `0 -> radiswap{...}; 1 -> transfer_xrd{...}`? That solves the above problems, and the contained transactions are easily identifiable anyways, and AFAIU we only list all these scenario executions (fully) from the endpoint (i.e. no-one queries them by any ID). Do you see any downsides of "scenario order number"?

Reply:
Sure - that sounds good :)